### PR TITLE
Add Twitter username to user profile

### PR DESCRIPTION
**Pull request by LoChanHi5**
Issue: https://github.com/rubygems/rubygems.org/issues/1302

Add optional Twitter handle as a link on user profile page.
screen shot 2016-07-01 at 11 43 28 pm
screen shot 2016-07-01 at 11 44 14 pm

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Great start!

Let's use `twitter_username` and "Twitter username" instead of `handle` everywhere.

Also let's add a note on the edit profile page saying that this field is option and will be displayed publicly.

**Comment by dwradcliffe** [comment]
Also maybe we should be explicit about the format (with or without the `@`) or normalize it after save?

**Comment by LoChanHi5** [comment]
@dwradcliffe I added instructions on the field and was explicit about not including `@`. The instructions will need to be translated though.

**Comment by LynnCo** [comment]
I usually do both! Be explicit about the formatting by adding the @ before the input field ([example](http://getbootstrap.com/components/#input-groups-basic) and then also removing the @ after save if its present.

File: app/views/profiles/edit.html.erb [hunk 1]
```diff
@@ -17,6 +17,20 @@
...
+    <%= form.label :twitter_username, :class => 'form__label' %>
+
+    <p class='form__field__instructions'>
+      <%= t('.optional_public_field') %>
+    </p>
+
+    <%= 
+      form.text_field(:twitter_username,
+        placeholder: t('.twitter_username_format'),
+        :class => 'form__input')
```
Comment (dwradcliffe) [inline app/views/profiles/edit.html.erb]: Can you change this to new style hash?

**Comment by dwradcliffe** [comment]
Several places that have mixed or old-style hashrocket syntax. Let's use the new style hash wherever we add or change code.

**Comment by simi** [comment]
I think @ problem can be solved without using any text for translation, but via showing @ as part of the input.

screenshot-getbootstrap com 2016-07-03 16-31-29

**Comment by LoChanHi5** [comment]
@simi I like that idea. As for the translation, we still need to translate the 'Optional. Will be displayed publicly' part.

**Comment by LoChanHi5** [comment]
I'm not great with front-end development but I think it turned out alright. Feedback is welcomed. Also I didn't add a callback to filter out `@` before save because I feel the field is now intuitive enough for users to understand `@` is not needed. Also, we have validation in place to catch any invalid characters. Thoughts?

screen shot 2016-07-03 at 1 20 12 pm

**Comment by dwradcliffe** [comment]
Looking good! One last thing, on the profile page we should label the username as "Twitter: @example" or maybe add the twitter logo on that line?

**Comment by simi** [comment]
I think greyscale twitter birdie will match the same pattern as the email envelope above.

**Comment by dwradcliffe** [comment]
> I think greyscale twitter birdie will match the same pattern as the email envelope above.

@simi 👍  good call

**Comment by LoChanHi5** [comment]
screen shot 2016-07-06 at 9 33 53 pm
screen shot 2016-07-06 at 9 34 09 pm

**Comment by LoChanHi5** [comment]
screen shot 2016-07-06 at 9 35 38 pm
screen shot 2016-07-06 at 9 35 29 pm

**Comment by simi** [comment]
Looks nice!

I have just one question. I see for the envelope `'data-icon' => '✉'` is used. Isn't there chance to use that for twitter birdie also instead of displaying image? I see that `icomoon` font is used for that evelope and this birdie is available in `icomoon` basic set. Unfortunately I can't find icomoon source file in repo. Maybe @acacheung can help here.

**Comment by sonalkr132** [comment]
@simi Those are here: https://github.com/rubygems/rubygems.org/tree/master/public/fonts but I don't see twitter icon in the files.

**Comment by simi** [comment]
@sonalkr132 those are font files, but in icomoon you can save also json font "manifest" so you can edit your font later in icomoon app and add new glyphs (which should be handy here).

File: app/views/profiles/edit.html.erb [hunk 2]
```diff
@@ -17,6 +17,25 @@
   </div>
 
   <div class="text_field">
+    <%= form.label :twitter_username, class: 'form__label form__label__icon-container' do %>
+      <%=
+        image_tag("/images/twitter_icon.png", alt: 'Twitter icon', class: 'form__label__icon')
+      %>
+      
+      <span class='form__label__text'>@example</span>
```
Comment (dwradcliffe) [inline app/views/profiles/edit.html.erb]: Can you change this text to "Twitter username"?

**Comment by LoChanHi5** [comment]
@simi There doesn't seem to be a `selection.json` file (i.e., json font manifest) in the project directory. I'm not sure I can add the Twitter bird as a font to our existing font file without `selection.json`. Thoughts?

**Comment by dwradcliffe** [comment]
We can just use the image for now. If someone wants to work on a better solution we can do that in a new PR.

Looks great! Thanks @LoChanHi5 for this contribution!

**Comment by sonalkr132** [comment]
Congratulation on your first open source contribution :heart_eyes:

**Comment by LoChanHi5** [comment]
Thanks! I'm looking to contribute more. The hardest part is finding issues to work on

**Comment by sonalkr132** [comment]
> The hardest part is finding issues to work on

It gets easier :smile: May be you can complete #1200

**Comment by LoChanHi5** [comment]
@sonalkr132 Yay. Thanks for the tip!

**Comment by dwradcliffe** [comment]
@LoChanHi5 I noticed on staging that if the twitter username isn't set then the twitter logo still shows up (and it's in front of the edit profile link).

Screenshot:
rubygems org your community gem host 2016-07-11 12-20-34

Can you send another PR to fix that?

**Comment by LoChanHi5** [comment]
Oops. yes. wil fix

**Pull request merged**